hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -420,6 +454,7 @@ long acquireLock() throws UnknownHostException, TException, InterruptedException
     LockResponse lockResponse = metaClients.run(client -> client.lock(lockRequest));
     AtomicReference<LockState> state = new AtomicReference<>(lockResponse.getState());
     long lockId = lockResponse.getLockid();
+    tableLocksById.put(lockId, fullName);
```

Comment:
Maybe store the triplet, so we can close it:
- lockId
- fullName
- metaClients